### PR TITLE
Fix for GNU compiler

### PR DESCRIPTION
**Pull request by kaiyuan-cheng**
Revision to ensure compatibility with the GNU compiler.

### CODE REVIEW

**Review by bensonr — COMMENTED** [review]
What version of GNU was used for testing?  I ask because I'm confused about most of the changes in sfcsub.f.  Many of the statements that were changed are perfectly valid fortran and in many cases the same style is used in statements just preceding the ones that were altered.

**Comment by lharris4** [comment]
I have to agree with Rusty, although the new forms are easier to understand
there is nothing invalid about the older ones.

BTW will the new forms still pass-by-reference instead of making copies?

Thanks,
Lucas

**Comment by kaiyuan-cheng** [comment]
I used gfortran 8.3 and 10.2 for testing. Sorry that I don't remember 
what was going on back then. One thing I am sure is that those revisions 
does not change the solution.

**Comment by laurenchilutti** [comment]
@bensonr @lharris4 @kaiyuan-cheng - Is this PR okay to be approved?  I want to make sure it hasn't been forgotten.

File: gsmphys/sfcsub.F [hunk 1]
```diff
@@ -2684,13 +2684,22 @@ subroutine fixrdg(lugb,idim,jdim,fngrib,
       real (kind=kind_io8) gdata(idim*jdim)
       logical gaus
       real (kind=kind_io8) blno,blto
-      real (kind=kind_io8) data8(idim*jdim)
+      real (kind=kind_io8), allocatable :: data8(:)
       real (kind=kind_io4), allocatable :: data4(:)
 !
-      logical*1 lbms(mdata)
+      logical*1, allocatable :: lbms(:)
```
Comment (lharris4) [inline gsmphys/sfcsub.F:2690]: Explicit definitions, allocations, and initializations are always better than not. 

If we want to keep using the GFS Physics Driver moving forward then we may want to take the time to clean up the cruft and imprecise code within it.

File: gsmphys/wv_saturation.F [hunk 2]
```diff
@@ -726,7 +726,7 @@ function polysvp (T,typ)
        integer typ
 
 
-      if (1) then
+      if (.TRUE.) then
```
Comment (lharris4) [inline gsmphys/wv_saturation.F:729]: 1965 called they want their weakly-typed statements back.

**Review by lharris4 — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Pull request merged**